compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -732,21 +732,20 @@ where
         let layout = match this.variants {
             Variants::Single { index }
                 // If all variants but one are uninhabited, the variant layout is the enum layout.
-                if index == variant_index &&
-                // Don't confuse variants of uninhabited enums with the enum itself.
-                // For more details see https://github.com/rust-lang/rust/issues/69763.
-                this.fields != FieldsShape::Primitive =>
```

Comment:
Now that `!` and hence empty enums use `Variants::Empty`, this check does not seem to be needed any more. :)